packages/react-server/core/ClientController.js

```diff
@@ -169,30 +240,9 @@ class ClientController extends EventEmitter {
 		 *    History.events.POPSTATE: user clicked back button but browser didn't do a full page load
 		 *    History.events.PAGELOAD: full browser page load, not using History API.
 		 */
-		context.onNavigate( (err, page, path, type) => {
+		context.onNavigate( (err, page) => {
 			logger.debug('Executing navigate action');
 
-			// if this is a History.events.PUSHSTATE navigation, we should change the URL in the bar location bar
-			// before rendering.
-			// note that for browsers that do not have pushState, this will result in a window.location change
-			// and full browser load. It's kind of late to do that, as we may have waited for handleRoute to
-			// finish asynchronously. perhaps we should have an "URLChanged" event that happens before "NavigateDone".
-			if (type === History.events.PUSHSTATE && this._history) {
-				// See "Sorry folks", above.
-				if (!(history.state||{}).reactServerFrame){
-					this._history.replaceState({
-						reactServerFrame: pushFrame(),
-					}, null, location.path);
-				}
-				this._history.pushState({
-					reactServerFrame: pushFrame(),
-				}, null, path);
-			} else if (type === History.events.PAGELOAD && this._history && this._history.canClientNavigate()) {
-				this._history.replaceState({
-					reactServerFrame: pushFrame(),
-				}, null, path);
```

Comment:
Oh, just realized I introduced a bug here.  It's tough to see because this whole block of code had to _move_, but I also eliminated the `replaceState` on `PAGELOAD`.  My reasoning was that any client transition away from the initial page would handle the `replaceState`, so the one at load time seemed redundant.  I had forgotten about non-`react-server` users of the history navigation API. :hurtrealbad: